### PR TITLE
Add support for catkin_make_isolated. Enable source build using .rosinstall

### PR DESCRIPTION
**Pull request by 130s**
DO NOT MERGE YET.

This addresses #21.

<s>This should be considered as a work-in-progress; I might need to ass more commits to let it pass the travis test of this repository.</s>

### CODE REVIEW

**Comment by 130s** [comment]
Now good for review.
@shaun-edwards @ipa-mdl

File: travis.sh [hunk 1]
```diff
@@ -65,6 +65,7 @@ function error {
 }
 
 BUILDER=catkin
```
Comment (ipa-mdl) [inline travis.sh]: This line overwrite the settings from .travis.yml 
Try `${BUILDER:=catkin}`

**Comment by 130s** [comment]
@ipa-mdl your comment is addressed. And I thought I included @gavanderhoorn but I didn't, sorry.

Currently test is failing due to the test errors in the downstream repos. I'm not sure if the [repositories I added](https://github.com/130s/industrial_ci/commit/04b1e7e30c73cbc1a2e5b52370c43fd1110834db) for testing are the most appropriate...

**Comment by shaun-edwards** [comment]
+0...I don't really know enough to verify the change, but I support it.

**Comment by 130s** [comment]
@shaun-edwards thanks for the review.

Sorry I found that this isn't ready for merge yet; I need to resolve the test failure.

**Comment by 130s** [comment]
To test this repository I'm using `industrial_core`. 
Only the test that uses `catkin_make_isolated` seems to be failing in testing `simple_message` ([here](https://travis-ci.org/ros-industrial/industrial_ci/jobs/113992035#L5270)). I see `industrial_core` [now passes Travis](https://github.com/ros-industrial/industrial_core/pull/136) so I'm not sure why it still fails here.

**Comment by shaun-edwards** [comment]
@130s, not sure if this is the issue, but `industrial_core` cannot be run as concurrent jobs.  Turn this off [here](https://travis-ci.org/ros-industrial/industrial_ci/settings).

**Comment by ipa-mdl** [comment]
I think we can add this more easily after the split-up (#98)

**Comment by ipa-mdl** [comment]
This is heavily outdated, I will close this now.
`catkin_make_isolated` is now supported in #361

**Pull request closed without merging**